exec/java-exec/src/test/java/org/apache/drill/exec/fn/impl/TestCastFunctions.java

```diff
@@ -78,4 +81,71 @@ public void testToDateForTimeStamp() throws Exception {
         .build()
         .run();
   }
+
+  @Test // DRILL-4970
+  public void testCastNegativeFloatToInt() throws Exception {
+    try {
+      test("create table dfs_test.tmp.table_with_float as\n" +
```

Comment:
Also, would be great if this can be unit tested directly, without a Drillbit. Use the `SubOperatorTest` framework if needed. But, since these are simple math functions, a basic JUnit test should be fine. Allows testing many more values without the complexity of a query.